src/main/java/globalquake/ui/stationselect/SearchBar.java

```diff
@@ -0,0 +1,368 @@
+            search = "";
+            setText("");
+        }
+        else{
+            image = ((javax.swing.ImageIcon) iconSearch).getImage();
+        }
+        g2.drawImage(image, width-height+marginImage + 3, marginButton+marginImage, imageSize, imageSize, null);
+        g2.dispose();
+    }
+
+    @Override
+    public void paint(Graphics g) {
+        super.paint(g);
+        if (getText().length() == 0) {
+            int h = getHeight();
+            ((Graphics2D)g).setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING,RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
+            Insets ins = getInsets();
+            FontMetrics fm = g.getFontMetrics();
+            int c0 = getBackground().getRGB();
+            int c1 = getForeground().getRGB();
+            int m = 0xfefefefe;
+            int c2 = ((c0 & m) >>> 1) + ((c1 & m) >>> 1);
+            g.setColor(new Color(c2, true));
+            g.drawString(hintText, ins.left, h / 2 + fm.getAscent() / 2 - 2);
+        }
+    }
+    private float getAlpha(){
+        float width = getWidth()/2;
+        float alpha = (location)/(-width);
+        alpha+=1;
+        if(alpha<0){
+            alpha = 0f;
+        }
+        if(alpha>1){
+            alpha = 1f;
+        }
+        return alpha;
+    }
+    private boolean checkMouseOver(Point mouse){
+        if(mouse==null){
+            return false;
+        }
+        int width = getWidth();
+        int height = getHeight();
+        int buttonSize = height - 5 *2;
+        Point point = new Point(width-height+3, 5);
+        Ellipse2D.Double circle = new Ellipse2D.Double(point.x, point.y, buttonSize, buttonSize);
+        return circle.contains(mouse);
+    }
+
+    private void parseSearch(String search){
+        try{
+            JSONParser parser = new JSONParser();
+            JSONObject rootObject = (JSONObject) parser.parse(new FileReader("src/main/resources/polygons/countriesHD.json"));
+
+            JSONArray countryData = (JSONArray) rootObject.get("features");
+           
+            String userInput = search.toLowerCase();
+            userInput = userInput.substring(0, 1).toUpperCase() + userInput.substring(1);
+
+            for(Object obj: countryData){
+                JSONObject countryObject = (JSONObject) obj;
+                JSONObject properties = (JSONObject) countryObject.get("properties");
+                String countryName = (String) properties.get("name");
+                JSONObject geoObject = (JSONObject) countryObject.get("geometry");
+
+                GeoJsonObject o = new ObjectMapper().readValue(geoObject.toJSONString(), GeoJsonObject.class);
+
+                if(countryName.equals(userInput)){
+                    System.out.println("Found country: " + countryName);
+                    countryFound = true;
+                    foundCoordinates = o;
+                }
+
+            }
+        }
+        catch(Exception e){
+            System.out.println("Error: " + e);
+        }
```

Comment:
Within this some sort of output must be shown to the user indicating that the country searched is a valid search. Currently the success is displayed in the terminal.